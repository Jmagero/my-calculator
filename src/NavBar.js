/* eslint-disable */
import React from "react";
import { NavLink } from "react-router-dom"
const NavBar = () => {
    const links = [
        {
            id: 1,
            path: '/Home',
            text: 'Home',
        },

        {
            id: 2,
            path: "/",
            text: "Calculator"
        },

        {
            id: 3,
            path: '/Quote',
            text: 'Quote',
        }
    ]
    return(
        <React.Fragment>
            <nav
            style={{
                display: 'flex',
                marginBottom: '2rem',
                justifyContent: "space-between"
            }}>
                <h1
                style= {{
                    fontSize: "3rem",
                    fontWeight: "600",
                    lineHeight: '1em',
                    color: "#000",
                    textAlign: "left"
                }}>
                    Math Magicians 
                </h1>
                <ul className="nav-links">
                    {links.map(link => {
                      return (
                        <li key={link.id}
                        style={{
                            padding: "0  0.5rem",
                        }}>
                            <NavLink to={link.path} activeClassName="active-link" exact>
                                {link.text}
                            </NavLink>
                        </li>
                      )
                    })}
                </ul>
            </nav>
        </React.Fragment>
    )
}

export default NavBar;