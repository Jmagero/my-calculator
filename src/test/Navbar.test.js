import React from 'react';
import renderer from 'react-test-renderer';
import {screen, render} from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../NavBar';

it('renders NavBar', () => {
    const tree = renderer.create(NavBar)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders Home link in NavBar component', () => {
    render(
    <BrowserRouter>
       <NavBar />
    </BrowserRouter>
    );
    expect(screen.getByText('Home')).toBeInTheDocument();
});

it('renders Quote link in NavBar component', () => {
    render(
        <BrowserRouter>
         <NavBar />
        </BrowserRouter>
     );
    expect(screen.getByText('Quote')).toBeInTheDocument();
});
