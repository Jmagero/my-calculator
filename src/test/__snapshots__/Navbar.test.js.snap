// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBar 1`] = `null`;
